pandas/core/construction.py

```diff
@@ -531,7 +531,7 @@ def sanitize_array(
         dtype = dtype.numpy_dtype
 
     # extract ndarray or ExtensionArray, ensure we have no PandasArray
-    data = extract_array(data, extract_numpy=True)
+    data = extract_array(data, extract_numpy=True, extract_range=True)
```

Comment:
I think this is the correct place to handle RangeIndexes, but this leaves us in the same position with other array types, like dask arrays, hence the copy below